src/coreclr/vm/jitinterface.cpp

```diff
@@ -11950,10 +11993,23 @@ void* CEEJitInfo::getFieldAddress(CORINFO_FIELD_HANDLE fieldHnd,
 
         GCX_COOP();
 
-        base = (void *) field->GetBase();
+        // Check if the field holds a frozen boxed static, return its content in that case
+        result = OBJECTREFToObject(getFrozenBoxedStatic(field));
```

Comment:
This still has the race condition. Consider situation where some other threads runs the static constructor between the point where `CORINFO_FLG_FIELD_STATIC_IN_HEAP` flag is returned and the address is computed here.

Reply:
Ah, right. Would it better to return address of the boxed static directly in getFieldInfo then?

Reply:
Yes, that should work.